spec/controllers/challenge_claims_controller_spec.rb

```diff
@@ -25,6 +25,59 @@
       it_redirects_to_with_error(root_path, \
                                  "You aren't allowed to see that user's claims.")
     end
+
+    context "for a prompt meme" do
+      let(:signup) { create(:prompt_meme_signup) }
```

Comment:
Surely it gets evaluated when we invoke `collection` in `get :index, params: { collection_id: collection.name }`? That's when `collection` gets evaluated, and `collection` is taken from `signup`, which is context-dependent. We want it to be promptmeme-specific, rather than a `challenge_signup`.

What I want from this PR is to fix the originally reported bug and be correct.

I don't want it to be perfect, so I'd rather not go back and fix every little suboptimal bit of code, or we'll be here all year.

Reply:
I was confused too because I didn't see `signup` being used... but I _think_ I get it now. 

On line 9, we have `let(:collection) { signup.collection }`. The `signup` that uses is the one on line 8, `let(:signup) { create(:challenge_signup) }`.

The idea here is that when `collection` is invoked with `get :index, params: { collection_id: collection.name }`,  it will use _this_ `let(:signup) { create(:prompt_meme_signup) }` instead of the more generic one on line 8, right?

Reply:
Yup. At least that's my understanding of lazy evaluation in tests.

The boxes I don't want to look under within the scope of this PR are:
1. Does it _actually_ make a difference whether we factory-create a prompt meme or gift exchange for this test? (The factories default to gift exchange for most challenge-related entities.)
1. If not, why? (Should the factories be more complete?)
1. During real life use, does the challenge claims controller even receive any other types of challenges than prompt memes?

I have now set up this branch for challenge claims mop-up:
https://github.com/CristinaRO/otwarchive/tree/AO3-TBD-challenge-claims-cleanup
Collaboration welcome!

Reply:
Sounds good to me, I missed the line `let(:collection) { signup.collection }`.